src/python/test/test_dxclient.py

```diff
@@ -7463,44 +7463,44 @@ def test_get_app_omit_resources(self):
                     break
             self.assertTrue(seenResources)
 
-    @unittest.skipUnless(testutil.TEST_ENV, 'skipping test that would clobber your local environment')
-    @unittest.skipUnless(testutil.TEST_ISOLATED_ENV, 'skipping test that would create apps')
-    @unittest.skipUnless(testutil.TEST_MULTIPLE_USERS, 'skipping test that would require another user')
+    @unittest.skipUnless(testutil.TEST_ISOLATED_ENV and testutil.TEST_MULTIPLE_USERS,
+                         'skipping test that would create apps and another user')
     def test_uninstall_app(self):
-        second = json.loads(os.environ['DXTEST_SECOND_USER'])
-        second_user_id = second['user']
+        second_user_id = json.loads(os.environ['DXTEST_SECOND_USER'])['user']
         authorized_users = [second_user_id]
         name = 'uninstall_test'
         app_spec = {
             "name": name,
             "title": name,
             "dxapi": "1.0.0",
             "runSpec": {"file": "code.py", "interpreter": "python2.7"},
-            "inputSpec": [{"name": "in1", "class": "file"}],
-            "outputSpec": [{"name": "out1", "class": "file"}],
+            "inputSpec": [],
+            "outputSpec": [],
             "description": "Description\n",
             "developerNotes": "Developer notes\n",
             "authorizedUsers": authorized_users,
             "openSource": True,
             "version": "0.0.1"
             }
         app_dir = self.write_app_directory(name, json.dumps(app_spec), "code.py", code_content="import os\n")
-        os.mkdir(os.path.join(app_dir, "resources"))
-        with open(os.path.join(app_dir, "resources", "resources_file"), 'w') as f:
-            f.write('content\n')
         build_cmd = "dx build --create-app --json --publish "
         app_json = json.loads(run(build_cmd + app_dir))
         self.assertEqual(app_json['name'], name)
         run("dx install " + name, env=as_second_user())
         output = json.loads(run("dx find apps --installed --json", env=as_second_user()))
         self.assertIn(name, [x['describe']['name'] for x in output])
         dxpy.api.app_remove_authorized_users(app_json['id'],
-                                                      input_params={'authorizedUsers': list(authorized_users)})
+                                             input_params={'authorizedUsers': list(authorized_users)})
         output = dxpy.api.app_describe(app_json['id'])
         self.assertNotIn(second_user_id, output['authorizedUsers'])
+        dxpy.api.project_invite(self.project, input_params={'invitee': second_user_id, 'level': 'VIEW'})
```

Comment:
dx probably guesses that the thing being described could be the name of a data object and tries to look for it. This is probably something where we have room for improvement, especially when the thing being described is obviously a user, app, etc.

However, the command below is run from the perspective of the second user, and I believe that for that user the project context points to a project for which they don't have VIEW access (since it was created solely for user1). That's a situation that is very likely to cause weird things to happen (especially in "describe"), so some readable project should probably be set explicitly.